src/content-steering-controller.js

```diff
@@ -120,46 +122,85 @@ export default class ContentSteeringController extends videojs.EventTarget {
 
     // trigger a steering event if we have a pathway from the content steering tag.
     // this tells VHS which segment pathway to start with.
-    if (this.defaultPathway) {
+    // If queryBeforeStart is true we need to wait for the steering manifest response.
+    if (this.defaultPathway && !this.queryBeforeStart) {
       this.trigger('content-steering');
     }
+
+    if (this.queryBeforeStart) {
+      this.requestSteeringManifest(this.queryBeforeStart);
+    }
   }
 
   /**
    * Requests the content steering manifest and parse the response. This should only be called after
    * assignTagProperties was called with a content steering tag.
+   *
+   * @param {boolean} withQueryBeforeStart If true, the request should be made with the initial serverUri.
+   *    This scenario is specific to DASH when the queryBeforeStart parameter is true.
+   *    This scenario should only happen once on initalization.
    */
-  requestSteeringManifest() {
+  requestSteeringManifest(withQueryBeforeStart = false) {
```

Comment:
Do we need this argument? I see the only usage with `this.queryBeforeStart`, but since this info is available in `this`, we can use it directly in the method

Reply:
The reason we use this param instead of just using `queryBeforeStart` is because we only want to use that specific reload uri when we call `requestSteeringManifest` during the initial parsing of the manifest file. We will call `requestSteeringManifest` many times later while `queryBeforeStart` is true, but we only want it to use the unparameterized reload uri before the first request to the content steering server.

Reply:
Given the following code:

```js
    if (this.queryBeforeStart) {
      this.requestSteeringManifest(this.queryBeforeStart);
    }
```    

And the fact that `this.queryBeforeStart` is a boolean, I believe it is more semantically right to pass URI instead of this boolean again:

```js
 if (this.queryBeforeStart) {
     const reloadUri = this.getRequestURI(this.steeringManifest.reloadUri)
      this.requestSteeringManifest(reloadUri);
    }
``` 

And for the method declaration part:
`requestSteeringManifest(reloadUri = this.steeringManifest.reloadUri)`